### PR TITLE
Add custom claim support

### PR DESCRIPTION
**Pull request by liubog2008**
Based on #1

### CODE REVIEW

**Comment by liubog2008** [comment]
@ddysher

File: Dockerfile [hunk 1]
```diff
@@ -18,6 +18,6 @@ COPY _output/bin/dex /usr/local/bin/dex
 COPY web /web
 WORKDIR /
 
-ENTRYPOINT ["dex"]
+#ENTRYPOINT ["dex"]
```
Comment (deyuanix) [inline Dockerfile:21]: revert?

File: connector/cauth/cauth.go [hunk 2]
```diff
@@ -234,3 +246,61 @@ func (c *cauthConnector) auth(username string, password string) (*v2.User, bool,
...
+	}
+
+	u, err := c.getUser(data.UserID)
+	if err != nil {
+		return ident, fmt.Errorf("cauth: can't get user info, user %v does not exists", data.UserID)
+	}
+
+	if claim, err := c.getClaim(data.UserID); err != nil {
+		return connector.Identity{}, err
+	} else {
```
Comment (deyuanix) [inline connector/cauth/cauth.go:291]: nit: do not use else here

File: connector/cauth/cauth.go [hunk 3]
```diff
@@ -234,3 +246,61 @@ func (c *cauthConnector) auth(username string, password string) (*v2.User, bool,
...
+		return connector.Identity{
+			UserID:        u.Username,
+			Username:      u.Nickname,
+			Email:         u.Email,
+			EmailVerified: u.EmailVerified,
+			Groups:        claim.Groups,
+			CustomClaims: map[string]interface{}{
+				"teams": claim.Teams,
```
Comment (deyuanix) [inline connector/cauth/cauth.go:299]: do we need `tenant` claim?

**Comment by deyuanix** [comment]
BTW, this looks like the title should be refresh token support?

File: connector/cauth/cauth.go [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+	} else if err := json.Unmarshal(body, &u); err != nil {
+		return nil, err
+	} else {
+		return &u, nil
+	}
+}
+
+func (c *cauthConnector) Refresh(ctx context.Context, s connector.Scopes, ident connector.Identity) (connector.Identity, error) {
+	c.logger.Infof("connector data %v", ident)
+	if len(ident.ConnectorData) == 0 {
```
Comment (deyuanix) [inline connector/cauth/cauth.go:275]: why `len(ident.ConnectorData) == 0` means session expired?

File: connector/cauth/cauth.go (hunk 3 above)
Comment (liubog2008) [reply connector/cauth/cauth.go:299]: yes, I'll add it after cauth is ok

**Pull request closed without merging**